renderdoc/driver/gl/gl_driver.cpp

```diff
@@ -2398,6 +2398,15 @@ void WrappedOpenGL::SwapBuffers(void *windowHandle)
       {
         auto remove = wit;
         ++wit;
+
+        // if we are about to remove a window because it is too old, let's
+        // check if it is the active window and update if so
+        if(RenderDoc::Inst().IsActiveWindow(cit->first, remove->first))
+        {
+          RenderDoc::Inst().SetActiveWindow(ctxdata.ctx, windowHandle);
+          activeWindow = true;
+        }
```

Comment:
This is good and it will work, but I think we could perhaps make a better fix. In my current code I don't call `RemoveFrameCapturer` for these windows being deleted, so they end up lying around. On platforms where you can change the active window it means you'd have to cycle through these 'invisible' windows lying around which aren't active anymore.

If you call `RenderDoc::Inst().RemoveFrameCapturer` here without any if check, then inside that function it makes sure that removing the active window then shifts to another context/window - which in your case would be the VR window.